core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -872,6 +900,20 @@ private RelNode mockCountOver(String table,
     return rel;
   }
 
+  @Test void testHashDistributionWithoutKeys() {
+    final RelNode root = createSortPlan(RelDistributions.hash(new ArrayList<>()));
+    final RelJsonWriter writer = new RelJsonWriter();
```

Comment:
Use `Collections.emptyList()` instead of `new ArrayList<>()`.

Reply:
> Use `Collections.emptyList()` instead of `new ArrayList<>()`.

Thanks for your suggestion, updated already.